src/Asset/EntrypointLookup.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\WebpackEncoreBundle\Asset;
+
+/**
+ * Returns the CSS or JavaScript files needed for a Webpack entry.
+ *
+ * This reads a JSON file with the format of Webpack Encore's entrypoints.json file.
+ * @final
+ */
+class EntrypointLookup
+{
+    private $entrypointJsonPath;
+
+    private $entriesData;
+
+    private $returnedFiles = [];
+
+    public function __construct(string $entrypointJsonPath)
+    {
+        $this->entrypointJsonPath = $entrypointJsonPath;
+    }
+
+    public function getJavaScriptFiles(string $entryName): array
+    {
+        return $this->getEntryFiles($entryName, 'js');
+    }
+
+    public function getCssFiles(string $entryName): array
+    {
+        return $this->getEntryFiles($entryName, 'css');
+    }
+
+    private function getEntryFiles(string $entryName, string $key): array
+    {
+        $this->validateEntryName($entryName);
+        $entriesData = $this->getEntriesData();
+        $entryData = $entriesData[$entryName];
+
+        if (!isset($entryData[$key])) {
+            throw new \InvalidArgumentException(sprintf('Could not find "%s" key for the "%s" entry.', $key, $entryName));
```

Comment:
if an entry point does not have JS or CSS, should we treat it as an error, or use an empty list ?

Reply:
Good catch. But I just double-checked this: that IS the behavior: if there is no css files or js files for an entry, an empty array is returned (just added a test for this). This is checking if the `js` or `css` key is completely missing from that entry in `entrypoints.json`, which is not possible (so would be some bad situation, or someone using a custom `entrypoints.json` that is not valid.